cpp/src/arrow/record_batch.h

```diff
@@ -234,6 +234,67 @@ class ARROW_EXPORT RecordBatchReader {
     return batch;
   }
 
+  class RecordBatchReaderIterator {
```

Comment:
Nit, but since this is a nested class, perhaps calling it `Iterator` is sufficient?

Reply:
In https://github.com/apache/arrow/blob/d88e23273fd4eb7945a5fb94cfdb6315f412ea83/cpp/src/arrow/record_batch.cc#L369 the name `Iterator` is in conflict.

Reply:
Other options: 
- `RangeIterator` as in https://github.com/apache/arrow/blob/d88e23273fd4eb7945a5fb94cfdb6315f412ea83/cpp/src/arrow/util/iterator.h#L134

Reply:
Ah, fair enough. We can keep it like this, then.